ext/mbstring/libmbfl/mbfl/mbfilter.c

```diff
@@ -312,6 +312,14 @@ int mbfl_encoding_detector_feed(mbfl_encoding_detector *identd, mbfl_string *str
 	unsigned char *p = string->val;
 	int bad = 0;
 
+	for (int i = 0; i < num; i++) {
+		mbfl_convert_filter *filter = identd->filter_list[i];
+		mbfl_encoding_detector_data *data = &identd->filter_data[i];
+		if (identd->strict && filter->from->check != NULL && !(filter->from->check)(p, n)) {
+			data->num_illegalchars++;
```

Comment:
You also need to increment `bad`, or the below loop won't break when N-1 candidate encodings are eliminated and we are in non-strict detection mode.

Reply:
Hmm, it's a bit worrying that this wasn't caught by unit tests.

Reply:
I think your comment is not correct, but thanks to that I realized the problem with this for statement itself.
This for statement does nothing in non-strict detection mode, it just defines variables.
I would prefer to enclose this for statement itself in an if statement.

Reply:
Do you think the following loop will still break early in non-strict mode if N-1 encodings are rejected and only one remains? If so, why?

Reply:
Hmmm, maybe I'm misunderstanding something, but in non-strict mode the encoding should not be rejected by my added code.
This is due to our decision not to use the result of the check function in the non-strict detection mode of mb_detect_encoding.
In other words, I don't think the situation "in non-strict mode if N-1 encodings are rejected and only one remains" would happen in the first place.